app/components/submission/SubmissionPage.js

```diff
@@ -27,64 +38,100 @@ const FormStep = ({ history, updateSubmission, nextUrl, ...props }) => (
   />
 )
 
-const SubmissionPage = ({ match, history }) => (
-  <WithCurrentSubmission>
-    {(updateSubmission, initialValues) => (
-      <Switch>
-        <Route
-          path={`${match.path}/upload`}
-          render={() => (
-            <FormStep
-              component={FileUploadsPage}
-              history={history}
-              initialValues={initialValues}
-              nextUrl={`${match.path}/metadata`}
-              updateSubmission={updateSubmission}
-              validationSchema={fileUploadsSchema}
-            />
-          )}
-        />
-        <Route
-          path={`${match.path}/metadata`}
-          render={() => (
-            <FormStep
-              component={ManuscriptMetadata}
-              history={history}
-              initialValues={initialValues}
-              nextUrl={`${match.path}/suggestions`}
-              updateSubmission={updateSubmission}
-              validationSchema={manuscriptMetadataSchema}
-            />
-          )}
-        />
-        <Route
-          path={`${match.path}/suggestions`}
-          render={() => (
-            <FormStep
-              component={ReviewerSuggestions}
-              history={history}
-              initialValues={initialValues}
-              nextUrl="/dashboard"
-              updateSubmission={updateSubmission}
-              validationSchema={reviewerSuggestionsSchema}
-            />
-          )}
-        />
-        <Route
-          render={() => (
-            <FormStep
-              component={AuthorDetailsPage}
-              history={history}
-              initialValues={initialValues}
-              nextUrl={`${match.path}/upload`}
-              updateSubmission={updateSubmission}
-              validationSchema={authorDetailsSchema}
-            />
-          )}
-        />
-      </Switch>
-    )}
-  </WithCurrentSubmission>
-)
+const SubmissionPage = ({ match, history }) => {
+  // TODO update interval in ms - define this somewhere else
+  const updateInterval = 5000
```

Comment:
Open to suggestions for this one.

Reply:
it could be a prop of the wrapper you are making?

Reply:
That's exactly what I did, I moved everything related to this data saving on timer to one place only.